gossipd/routing.c

```diff
@@ -538,21 +567,71 @@ static bool check_channel_announcement(
 	       check_signed_hash(&hash, bitcoin2_sig, bitcoin2_key);
 }
 
-/* While master always processes in order, bitcoind is async, so they could
- * theoretically return out of order. */
-static struct pending_cannouncement *
-find_pending_cannouncement(struct routing_state *rstate,
-			   const struct short_channel_id *scid)
+struct routing_channel *routing_channel_new(const tal_t *ctx,
+					    struct short_channel_id *scid)
 {
-	struct pending_cannouncement *i;
+	struct routing_channel *chan = tal(ctx, struct routing_channel);
+	chan->scid = *scid;
+	chan->connections[0] = chan->connections[1] = NULL;
+	chan->nodes[0] = chan->nodes[1] = NULL;
+	chan->txout_script = NULL;
+	chan->state = TXOUT_FETCHING;
+	chan->public = false;
+	memset(&chan->msg_indexes, 0, sizeof(chan->msg_indexes));
+	return chan;
+}
 
-	list_for_each(&rstate->pending_cannouncement, i, list) {
-		if (short_channel_id_eq(scid, &i->short_channel_id))
-			return i;
-	}
-	return NULL;
+static void remove_connection_from_channel(struct node_connection *nc,
+					   struct routing_state *rstate)
+{
+	struct routing_channel *chan = uintmap_get(
+	    &rstate->channels, short_channel_id_to_uint(&nc->short_channel_id));
+	struct node_connection *c = chan->connections[nc->flags & 0x1];
+	status_trace("Removing node_connection from channels map");
```

Comment:
The duplicate messages feel strange.  Would it be better?

    if (c == NULL) {
        status_trace("Removing node_connection from channels map: NOT FOUND");
        return;
    }

    status_trace("Removing node_connection from channels map: FOUND");

Reply:
Ooops, that was not intended to be in there at all, just had a hard time debugging this.